api/types/events/events.go

```diff
@@ -2434,3 +2434,15 @@ func (m *WorkloadIdentityDelete) TrimToMaxSize(_ int) AuditEvent {
 func (m *GitCommand) TrimToMaxSize(_ int) AuditEvent {
 	return m
 }
+
+func (m *AWSICResourceSync) TrimToMaxSize(maxSize int) AuditEvent {
+	return m
+}
+
+func (m *AWSICPermissionAssignment) TrimToMaxSize(_ int) AuditEvent {
+	return m
+}
+
+func (m *AWSICPrincipalProvisioning) TrimToMaxSize(_ int) AuditEvent {
+	return m
+}
```

Comment:
any of these that contain repeated fields seem like exactly the type of event that might end up being very big, where it should be mandatory to implement TrimToMaxSize. if the events don't fit in a gRPC message the event handler cannot handle them

Reply:
Good point. I ended up not importing detailed resource id and ARNs so only status field is imported now. https://github.com/gravitational/teleport/pull/52059